lib/src/Base/Algo/ApproximationAlgorithmImplementation.cxx

```diff
@@ -211,6 +216,11 @@ Bool ApproximationAlgorithmImplementation::getVerbose() const
   return verbose_;
 }
 
+Bool ApproximationAlgorithmImplementation::getIsModelSelection() const
+{
+  return isModelSelection_;
+}
+
```

Comment:
Suggestion: implement `IsModelSeleciton()` and return `true`. Do not use the attribute.

Reply:
Ok. Fixed at 8d8ebe32519c39c30f3989424581f6336673145e. The updated code at [ApproximationAlgorithmImplementation.cxx#L211](https://github.com/mbaudin47/openturns/blob/d2d18e244254b79129c880b1857a7c416ff819f3/lib/src/Base/Algo/ApproximationAlgorithmImplementation.cxx#L211) throws an exception. Children classes override this method and return a hard-coded value. See for example [LeastSquaresMetaModelSelection.cxx#L213](https://github.com/mbaudin47/openturns/blob/d2d18e244254b79129c880b1857a7c416ff819f3/lib/src/Base/Algo/LeastSquaresMetaModelSelection.cxx#L213)